lib/http2/connection.c

```diff
@@ -887,7 +893,11 @@ static int handle_settings_frame(h2o_http2_conn_t *conn, h2o_http2_frame_t *fram
         if (ret != 0)
             return ret;
         { /* schedule ack */
-            h2o_iovec_t header_buf = h2o_buffer_reserve(&conn->_write.buf, H2O_HTTP2_FRAME_HEADER_SIZE);
+            h2o_iovec_t header_buf = h2o_buffer_try_reserve(&conn->_write.buf, H2O_HTTP2_FRAME_HEADER_SIZE);
```

Comment:
IIRC the upper size of `conn->_write.buf` is capped by something slightly above 64KB.